python_modules/dagster/dagster/_config/structured_config/typing_utils.py

```diff
@@ -1,8 +1,79 @@
-from typing import TYPE_CHECKING, Any, Optional, TypeVar, Union, cast
+from typing import TYPE_CHECKING, Any, Generic, Optional, Type, TypeVar, Union, cast
+
+import pydantic
+from typing_extensions import dataclass_transform, get_origin
+
+from .utils import safe_is_subclass
 
 if TYPE_CHECKING:
```

Comment:
Worthy of comment/explanation as to why this works while still having to do all the crazy shenanigans around temp classes.